content/docs/06-advanced/10-vercel-deployment-guide.mdx

```diff
@@ -0,0 +1,112 @@
+
+Note: if you run into the error "error: remote origin already exists.", this is because your local repository is still linked to the repository you cloned. To "unlink", you can run the following command:
+
+```bash
+rm -rf .git
+git init
+git add .
+git commit -m "init"
+```
+
+Rerun the code snippet from the previous step.
+
+## Import Project in Vercel
+
+On the [New Project](https://vercel.com/new) page, under the **Import Git Repository** section, select the Git provider that you would like to import your project from. Follow the prompts to sign in to your GitHub account.
+
+Once you have signed in, you should see your newly created repository from the previous step in the "Import Git Repository" section. Click the "Import" button next to that project.
+
+### Add Environment Variables
+
+Your application stores uses environment secrets to store your OpenAI API key using a `.env.local` file locally in development. To add this API key to your production deployment, expand the "Environment Variables" section and paste in your `.env.local` file. Vercel will automatically parse your variables and enter them in the appropriate `key:value` format.
+
+### Deploy
+
+Press the **Deploy** button. Vercel will create the Project and deploy it based on the chosen configurations.
+
+### Enjoy the confetti!
+
+To view your deployment, select the Project in the dashboard and then select the **Domain**. This page is now visible to anyone who has the URL.
+
+## Considerations
+
+When deploying an AI application, there are infrastructure-related considerations to be aware of.
+
+### Function Duration
+
+In most cases, you will call the large language model (LLM) on the server. By default, Vercel serverless functions have a maximum duration of 10 seconds on the Hobby Tier. Depending on your prompt, it can take an LLM more than this limit to complete a response. If the response is not resolved within this limit, the server will throw an error.
+
+You can specify the maximum duration of your Vercel function using [route segment config](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config). To update your maximum duration, add the following route segment config to the top of your route handler or the page which is calling your server action.
+
+```ts
+export const maxDuration = 30;
+```
+
+You can increase the max duration to 60 seconds on the Hobby Tier.
+
+## Security Considerations
+
+Given the high cost of calling an LLM, it's important to have measures in place that can protect your application from abuse.
+
+### Rate Limit
+
+Rate limiting is a method used to regulate network traffic by defining a maximum number of requests that a client can send to a server within a given time frame.
+
+Follow [this guide](https://vercel.com/guides/securing-ai-app-rate-limiting) to add rate limiting to your application.
+
+### Firewall
+
+A firewall helps protect your applications and websites from DDoS attacks and unauthorized access.
+
```

Comment:
What do you think about adding a troubleshooting / common issues section with links to the relevant troubleshooting pages